src/compiler/emitter.ts

```diff
@@ -7312,13 +7340,32 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 write("});");
             }
 
+            function lookupSpecifierName(declaration: ImportEqualsDeclaration | ImportDeclaration | ExportDeclaration): string {
+                let specifier: Node;
+                if (declaration.kind === SyntaxKind.ImportEqualsDeclaration) {
+                    specifier = (declaration as ImportEqualsDeclaration).moduleReference;
+                }
+                else {
+                    specifier = (declaration as ImportDeclaration | ExportDeclaration).moduleSpecifier;
+                }
+                let moduleSymbol = resolver.getSymbolAtLocation(specifier);
+                if (!moduleSymbol) {
+                    return undefined;
+                }
+                let moduleDeclaration = getDeclarationOfKind(moduleSymbol, SyntaxKind.SourceFile) as SourceFile;
+                if (!moduleDeclaration || isDeclarationFile(moduleDeclaration)) {
+                    return undefined;
+                }
+                return getExternalModuleNameFromPath(host, moduleDeclaration.fileName);
+            }
+
             interface AMDDependencyNames {
                 aliasedModuleNames: string[];
                 unaliasedModuleNames: string[];
                 importAliasNames: string[];
             }
 
-            function getAMDDependencyNames(node: SourceFile, includeNonAmdDependencies: boolean): AMDDependencyNames {
+            function getAMDDependencyNames(node: SourceFile, includeNonAmdDependencies: boolean, resolveModuleNames?: boolean): AMDDependencyNames {

```

Comment:
I do agree with @mhegazy that `resolveModuleNames` should have a better name as now it is not clear what difference will it make if I pass `true` instead of `false` here. Since `true` implies that module name will be replaced with its relative path from the current directory maybe it should be `emitRelativePathAsModuleName`?